prow/plank/reconciler.go

```diff
@@ -162,19 +165,35 @@ type reconciler struct {
 	opener             io.Opener
 	totURL             string
 	clock              clock.WithTickerAndDelayedExecution
-	serializationLocks *shardedLock
+	/* maxConcurrencySerializationLocks and jobQueueSerializationLocks are used to serialize
+	   reconciliation of ProwJobs that have concurrency limits that might affect eachother.
+
+	   The concurrency management strategy has 3 basic parts. Each part is skipped if the ProwJob
+	   does not specify a MaxConcurrency or JobQueueName.
+
+	   1. Serialize per the job and/or queue name as needed using these locks. This prevents
+	      concurrent reconciliation threads from triggering jobs beyond the concurrency limit.
+	   2. Compare against the ProwJob index to see how many jobs there are for the job and job queue
+	      and only trigger the job if it won't exceed the concurrency limit(s).
+	   3. Once the ProwJob is updated, wait until we see it updated in our cache before completing
+	      processing and releasing the serialization lock(s) acquired in step 1. This is necessary
+		  to prevent reconciliation threads from processing subsequent jobs before the ProwJob index
+		  used in step 2 is up to date.
```

Comment:
nit: indentation level on previous two lines.

Reply:
There was a mix of tabs and spaces that looked even in my editor, but not here. Fixed.